src/main/java/seedu/cafectrl/parser/Parser.java

```diff
@@ -295,6 +303,96 @@ private static Command prepareHelpCommand() {
         return new HelpCommand();
     }
 
+    /**
+     * Parses arguments in the context of the Delete command.
+     *
+     * @param menu menu of the current session
+     * @param arguments string that matches group arguments
+     * @return AddOrderCommand if command is valid, IncorrectCommand otherwise
+     */
+    private static Command prepareOrder(Menu menu, String arguments) {
+        final Pattern addOrderArgumentPatter = Pattern.compile(ADD_ORDER_ARGUMENT_STRING);
+        Matcher matcher = addOrderArgumentPatter.matcher(arguments);
+
+        // Checks whether the overall pattern of add order arguments is correct
+        if (!matcher.matches()) {
+            return new IncorrectCommand("Error: Incorrect format for the add order command.\n"
+                    + AddOrderCommand.MESSAGE_USAGE);
+        }
+
+        try {
+            // To retrieve specific arguments from arguments
+            String dishName = matcher.group(DISH_NAME_MATCHER_GROUP_NUM);
+            int dishQty = Integer.parseInt(matcher.group(ORDER_QTY_MATCHER_GROUP_NUM));
+
+            Dish orderedDish = getDishInMenu(dishName, menu);
+            if (orderedDish == null) {
+                return new IncorrectCommand(Messages.DISH_NOT_FOUND);
+            }
+
+            ArrayList<Ingredient> usedIngredientList = getIngredientList(orderedDish, dishQty);
+            float totalOrderCost = getDishPrice(orderedDish, dishQty);
+            Order order = new Order(dishName, dishQty, usedIngredientList, totalOrderCost);
```

Comment:
Perhaps you could simply pass in Order(dishName, dishQty) and let Order class calculate / retrieve the usedIngerdientList and totalOrderCost as they seem to be more of Order's responsibility rather than parser's?